src/Uno.UWP/Devices/Haptics/SimpleHapticsController.iOS.cs

```diff
@@ -1,48 +1,66 @@
-﻿#nullable enable
-
-using System;
-using System.Collections.Generic;
-using UIKit;
+public class HapticFeedbackManager
+{
+    private DateTime? lastClickTime = null;
 
-#nullable enable
+    public IReadOnlyList<SimpleHapticsControllerFeedback> SupportedFeedback { get; } = new SimpleHapticsControllerFeedback[]
+    {
+        new SimpleHapticsControllerFeedback(KnownSimpleHapticsControllerWaveforms.Click, TimeSpan.FromMilliseconds(100)),
+        new SimpleHapticsControllerFeedback(KnownSimpleHapticsControllerWaveforms.Press, TimeSpan.FromMilliseconds(300)),
+        new SimpleHapticsControllerFeedback(KnownSimpleHapticsControllerWaveforms.Click, TimeSpan.FromMilliseconds(50)), // Single Click
+        new SimpleHapticsControllerFeedback(KnownSimpleHapticsControllerWaveforms.Click, TimeSpan.FromMilliseconds(150)), // Double Click
+    };
 
-namespace Windows.Devices.Haptics
-{
-	public partial class SimpleHapticsController
-	{
-		public IReadOnlyList<SimpleHapticsControllerFeedback> SupportedFeedback { get; } = new SimpleHapticsControllerFeedback[]
-		{
-			new SimpleHapticsControllerFeedback(KnownSimpleHapticsControllerWaveforms.Click, TimeSpan.FromMilliseconds(100)),
-			new SimpleHapticsControllerFeedback(KnownSimpleHapticsControllerWaveforms.Press, TimeSpan.FromMilliseconds(300))
-		};
+    public void SendHapticFeedback(SimpleHapticsControllerFeedback feedback)
+    {
+        if (feedback is null)
+        {
+            throw new ArgumentNullException(nameof(feedback));
+        }
 
-		public void SendHapticFeedback(SimpleHapticsControllerFeedback feedback)
-		{
-			if (feedback is null)
-			{
-				throw new ArgumentNullException(nameof(feedback));
-			}
+        var impactStyle = FeedbackToImpactStyle(feedback);
+        using var impact = new UIImpactFeedbackGenerator(impactStyle);
+        impact.Prepare();
 
-			var impactStyle = FeedbackToImpactStyle(feedback);
-			using var impact = new UIImpactFeedbackGenerator(impactStyle);
-			impact.Prepare();
-			impact.ImpactOccurred();
-		}
+        if (feedback.Duration.TotalMilliseconds <= 100)
+        {
+            // Single Click
+            impact.ImpactOccurred();
+        }
+        else if (feedback.Duration.TotalMilliseconds <= 200)
+        {
+            // Potential Double Click - Check time interval with last click
+            if (lastClickTime.HasValue && (DateTime.Now - lastClickTime.Value).TotalMilliseconds < 200)
```

Comment:
@dr1rrb how do we determine that something is a double tap in gestures? We may want to align this.

Reply:
We are using this https://github.com/unoplatform/uno/blob/master/src/Uno.UWP/UI/Input/GestureRecognizer.cs#L27

But anyway, I'm not sure to understand the condition and why try to make distinction between simple click and double click. Here we end by doing `impact.ImpactOccurred();` anyway. Is it expected ?